include/conversation.php

```diff
@@ -1386,54 +1393,124 @@ function get_item_children($arr, $parent) {
 	return $children;
 }
 
-/// @TODO Add type-hint
-function sort_item_children($items) {
+/**
+ * Recursively sorts a tree-like item array
+ *
+ * @param array $items
+ * @return array
+ */
+function sort_item_children(array $items)
+{
 	$result = $items;
 	usort($result, 'sort_thr_created_rev');
 	foreach ($result as $k => $i) {
-		if (count($result[$k]['children'])) {
+		if (isset($result[$k]['children'])) {
 			$result[$k]['children'] = sort_item_children($result[$k]['children']);
 		}
 	}
 	return $result;
 }
 
-/// @TODO Add type-hint
-function add_children_to_list($children, &$arr) {
-	foreach ($children as $y) {
-		$arr[] = $y;
-		if (count($y['children'])) {
-			add_children_to_list($y['children'], $arr);
+/**
+ * Recursively add all children items at the top level of a list
+ *
+ * @param array $children List of items to append
+ * @param array $item_list
+ */
+function add_children_to_list(array $children, array &$item_list)
+{
+	foreach ($children as $child) {
+		$item_list[] = $child;
+		if (isset($child['children'])) {
+			add_children_to_list($child['children'], $item_list);
 		}
 	}
 }
 
-/// @TODO Add type-hint
-function conv_sort($arr, $order) {
-
-	if ((!(is_array($arr) && count($arr)))) {
-		return array();
+/**
+ * This recursive function takes the item tree structure created by conv_sort() and
+ * flatten the extraneous depth levels when people reply sequentially, removing the
+ * stairs effect in threaded conversations limiting the available content width.
+ *
+ * The basic principle is the following: if a post item has only one reply and is
+ * the last reply of its parent, then the reply is moved to the parent.
+ *
+ * This process is rendered somewhat more complicated because items can be either
+ * replies or likes, and these don't factor at all in the reply count/last reply.
+ *
+ * @param array $parent A tree-like array of items
+ * @return array
+ */
+function smart_flatten_conversation(array $parent)
```

Comment:
I'm slightly confused now because of the naming: I thought for new functions and methods "camelCase" should now be used? Please enlight me.

Reply:
Class method names are supposed to be "camelCase", standard function names are supposed to be "undescore_case".

Reply:
Okay, some day I probably will understand this.

Reply:
It just is a convention, to visually separate different symbol types. There's no understanding to be had, just getting used to.